### ISSUE TITLE
[tbendfelt on 2015-09-17 10:48:04] : pDAQ: LogSorter.py - Fix TCAL noise screening for Wild TCALs

### ISSUE THREAD

**Issue by dglo** [dglo/StringHub#265]
Issue 7445 Changed the way Wild TCALS are presented in logs. The noise filter needs to change accordingly.

Old Way:
>>>
stringHub#38 icecube.daq.rapcal.AbstractRAPCal-Processor-60A WARN [2015-09-08 17:10:43.587] Wild TCAL - cable len: 1.572284006906706E-5 avg. cable len: 1.6298741636506722E-5
<<<

New Way:
>>>
stringHub#2029 icecube.daq.domapp.dataprocessor.TCalProcessor-Processor-01B WARN [2015-09-09 14:55:20.694] Got RAPCal exception
icecube.daq.rapcal.RAPCalException: Wild TCAL - cable len: 1.563344460870607E-5 avg. cable len: 1.5622519853293056E-5
	at icecube.daq.rapcal.AbstractRAPCal.addIsochron(AbstractRAPCal.java:393)
	at icecube.daq.rapcal.AbstractRAPCal.update(AbstractRAPCal.java:336)
	at icecube.daq.domapp.dataprocessor.TCalProcessor.updateRapCal(TCalProcessor.java:131)
	at icecube.daq.domapp.dataprocessor.TCalProcessor.access$400(TCalProcessor.java:28)
	at icecube.daq.domapp.dataprocessor.TCalProcessor$EstablishedCALProcessor.process(TCalProcessor.java:202)
	at icecube.daq.domapp.dataprocessor.TCalProcessor.process(TCalProcessor.java:91)
	at icecube.daq.domapp.dataprocessor.SynchronousDataProcessor.process(SynchronousDataProcessor.java:94)
	at icecube.daq.domapp.dataprocessor.AsynchronousDataProcessor$4.call(AsynchronousDataProcessor.java:258)
	at icecube.daq.domapp.dataprocessor.AsynchronousDataProcessor$4.call(AsynchronousDataProcessor.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
<<<

**Comment by dglo** [comment]
[tbendfelt on 2015-09-17 11:18:22]
As a side effect, additional RAPCal Exceptions will also be filtered:

stringHub#53 icecube.daq.domapp.DataCollector-30A WARN [2015-03-10 10:12:09.100] Got RAPCal exception
icecube.daq.rapcal.RAPCalException
        at icecube.daq.rapcal.ZeroCrossingRAPCal.getFineTimeCorrection(ZeroCrossingRAPCal.java:23)
        at icecube.daq.rapcal.AbstractRAPCal.setupVierling(AbstractRAPCal.java:290)
        at icecube.daq.rapcal.AbstractRAPCal.access$000(AbstractRAPCal.java:11)
        at icecube.daq.rapcal.AbstractRAPCal$Isochron.<init>(AbstractRAPCal.java:37)
        at icecube.daq.rapcal.AbstractRAPCal.update(AbstractRAPCal.java:188)
        at icecube.daq.domapp.DataCollector.execRapCal(DataCollector.java:1101)
        at icecube.daq.domapp.DataCollector.runcore_interval(DataCollector.java:1669)
        at icecube.daq.domapp.DataCollector.runcore_universal(DataCollector.java:1358)
        at icecube.daq.domapp.DataCollector.run(DataCollector.java:1170)

**Comment by dglo** [comment]
[tbendfelt on 2015-09-17 11:19:27]
Filter "Got RAPCal exception" as noise.
------------------------------------------------------------------------
r15769 | bendfelt | 2015-09-17 11:16:23 -0500 (Thu, 17 Sep 2015) | 1 line
Changed paths:
   M /projects/dash/trunk/LogSorter.py

Issue 7532: Screen RAPCal exceptions.
------------------------------------------------------------------------

**Comment by dglo** [comment]
[tbendfelt on 2015-10-01 17:00:06]
There is a better way to fix this.  Restore the previous logging behavior for wild TCals (No stack trace).

This is targeted to Minocqua RC2.

**Comment by dglo** [comment]
[tbendfelt on 2015-10-01 17:24:11]
Second pass.  I restored the old logging behavior of wild Tcals so the old filtering logic will still work.

------------------------------------------------------------------------
r15792 | bendfelt | 2015-10-01 17:16:21 -0500 (Thu, 01 Oct 2015) | 1 line
Changed paths:
   M /projects/StringHub/trunk/src/main/java/icecube/daq/rapcal/AbstractRAPCal.java
   A /projects/StringHub/trunk/src/main/java/icecube/daq/rapcal/WildTCalException.java

Issue 7532: Use a specific exception subtype for wild tcals.
------------------------------------------------------------------------
r15793 | bendfelt | 2015-10-01 17:18:39 -0500 (Thu, 01 Oct 2015) | 1 line
Changed paths:
   M /projects/StringHub/trunk/src/main/java/icecube/daq/domapp/dataprocessor/TCalProcessor.java

Issue 7532: Do not log stack trace for wild tcals exceptions.
------------------------------------------------------------------------

------------------------------------------------------------------------
r15794 | bendfelt | 2015-10-01 17:20:49 -0500 (Thu, 01 Oct 2015) | 1 line
Changed paths:
   M /projects/dash/trunk/LogSorter.py

Issue 7532: Do not screen RAPCal exceptions.
------------------------------------------------------------------------
